docs/functional/architecture/0013-Scalar_Operators.md

```diff
@@ -0,0 +1,135 @@
+# Scalar operators
+
+How about "elementwise operators"?
+
+## What are scalar operators?
+
+Scalar operators are functions that take scalar arguments and return a scalar result. They can be used to implement arbitrary pointwise operations on fields.
+
+As an example, let's calculate the pointwise arithmetic-geometric mean of two fields:
+
+```python
+@scalar_operator
+def agm(x: float, y: float, tol: float) -> float:
+    an = x
+    gn = y
+    while abs(x - y) > tol:
+        an, gn = 0.5*(an + gn), math.sqrt(an * gn)
+    return an
+
+@field_operator
+def use_agm(x: Field[[D], float], y: Field[[D], float]) -> Field[[D], float]:
+    return agm(x, y)
+```
+
+In the code above, the function `agm` is executed once for every element pair of the `(x, y)` field pair, and a field of the same size as both `x` and `y` is produced. The function `agm` is essentially just a complicated pointwise binary operator, like `+` or `-`.
```

Comment:
I think it would also be good here to point out why this is different from a `scan_operator` . It is intuitive for a developer, but the same cannot be said for a regular user who just started to use both